spec/cordova/platform/addHelper.spec.js

```diff
@@ -16,34 +16,459 @@
+                    //require packge.json object
+                    spyOn(cordova_util, 'requireNoCache').and.returnValue(pkgJsonEmpty);
+                });
+
+                xit('should write out the platform just added/updated to the cordova.platforms property of package.json',function (done) {
+                    platform_addHelper('add', hooks_mock, projectRoot, ['atari'], {save: true}).then(function(result) {
+                        expect(fs.writeFileSync).toHaveBeenCalled();
+                    }).fail(function (err) {
+                        fail('unexpected failure handler invoked!');
+                        console.error(err);
+                    }).done(done);
+                });
+
+                xit('should only write the package.json file if it was modified', function (done) {
+                    var pkgJsonFull = { 'cordova': {'platforms': ['atari']}};
+                    cordova_util.requireNoCache.and.returnValue(pkgJsonFull);
+                    platform_addHelper('add', hooks_mock, projectRoot, ['atari'], {save: true}).then(function(result) {
+                        expect(fs.writeFileSync).not.toHaveBeenCalled();
+                    }).fail(function (err) {
+                        fail('unexpected failure handler invoked!');
+                        console.error(err);
+                    }).done(done);
+                });
+
+                xit('should file the after_platform_* hook',function (done) {
+                    platform_addHelper('add', hooks_mock, projectRoot, ['atari'], {save: true}).then(function(result) {
+                        expect(hooks_mock.fire).toHaveBeenCalledWith( 'before_platform_add', Object({ save: true, searchpath: undefined }));
+                    }).fail(function (err) {
+                        fail('unexpected failure handler invoked!');
+                        console.error(err);
+                    }).done(done);
+                });
+            });
+        });
+    });
+    describe('downloadPlatform', function () {
+        beforeEach(function () {
+            spyOn(Q, 'reject').and.callThrough();
+            platform_addHelper.downloadPlatform.and.callThrough();
+        });
+        describe('errors', function () {
+            it('should reject the promise should fetch fail', function (done) {
+                fetch_mock.and.returnValue(Q.reject('fetch has failed, rejecting promise'));
+                platform_addHelper.downloadPlatform(projectRoot, 'android', '67', {fetch: true}).then(function () {
+                    fail('success handler unexpectedly invoked');
+                }).fail(function (e) {
+                    expect(e.message).toContain('fetch has failed, rejecting promise');
+                }).done(done);
+            });
+
+            it('should reject the promise should lazy_load.git_clone fail', function (done) {
+                spyOn(events, 'emit');
+                spyOn(lazy_load, 'based_on_config').and.returnValue(false);
```

Comment:
I'd suggest moving these lazy_load spyOns into the beforeEach on line 328. Then just set the value as needed in the it via `lazy_load.based_on_config.and.returnValue(false);`